src/Compilers/CSharp/Portable/Syntax/NullableDirectiveMap.cs

```diff
@@ -91,6 +110,14 @@ private static ImmutableArray<(int Position, bool? State)> GetDirectives(SyntaxT
             return builder.ToImmutableAndFree();
         }
 
+        private static bool GetIsGenerated(SyntaxTree tree)
+        {
+            Func<SyntaxTrivia, bool> isComment = trivia => trivia.Kind() == SyntaxKind.SingleLineCommentTrivia || trivia.Kind() == SyntaxKind.MultiLineCommentTrivia;
```

Comment:
>isComment [](start = 37, length = 9)

Consider inlining this in the `IsGeneratedCode(...)` call below, so the compiler caches the instance.